### PR TITLE
feat: add all FME service URLs and pathParameters token substitution

### PR DESCRIPTION
**Pull request by rcooney**
Add all REST API URLs from https://docs.safe.com/fme/html/FME_REST/apidoc/v3/#.
Add Path Parameters input to perform token substitution on the Path.
fix: method parameter typo.

### CODE REVIEW

File: src/activities/UseFmeService.ts [hunk 1]
```diff
@@ -16,23 +16,214 @@ export interface UseFmeServiceInputs {
...
+        | "repositories/{repository}/items/{item}/parameters/{pubparam}"
+        | "repositories/{repository}/items/{item}/resources"
+        | "repositories/{repository}/items/{item}/resources/{resource}"
+        | "repositories/{repository}/items/{item}/services"
+        | "repositories/{repository}/items/{item}/services/{service}"
         | "resources/connections"
+        | "resources/connections/{resource}"
+        | "resources/connections/{resource}/copy/{path}"
+        | "resources/connections/{resource}/directdownloadurl/{path}"
+        | "resources/connections/{resource}/downloadzip/{directory}"
```
Comment (kenlyon) [inline src/activities/UseFmeService.ts:129]: Missing `"resources/connections/{resource}/directuploadurl/{path}"`

File: src/activities/UseFmeService.ts [hunk 2]
```diff
@@ -16,23 +16,214 @@ export interface UseFmeServiceInputs {
...
+        | "resources/connections/{resource}/download/{path}"
+        | "resources/connections/{resource}/filesys/{path}"
+        | "resources/connections/{resource}/move/{path}"
+        | "resources/connections/{resource}/directdownloadurlparameters"
```
Comment (kenlyon) [inline src/activities/UseFmeService.ts]: This should not have the `/connections/{resource}` path segment in it.

File: src/activities/UseFmeService.ts [hunk 3]
```diff
@@ -16,23 +16,214 @@ export interface UseFmeServiceInputs {
...
+        | "resources/connections/{resource}/directuploadurlparameters"
```
Comment (kenlyon) [inline src/activities/UseFmeService.ts]: This should not have the `/connections/{resource}` path segment in it.

File: src/activities/UseFmeService.ts [hunk 4]
```diff
@@ -16,23 +16,214 @@ export interface UseFmeServiceInputs {
...
+        | "resources/test/connections"
+        | "resources/test/connections/{resource}"
         | "resources/types"
         | "schedules"
         | "schedules/categories"
+        | "schedules/{category}"
```
Comment (kenlyon) [inline src/activities/UseFmeService.ts]: I didn't see this one in the docs.

File: src/activities/UseFmeService.ts [hunk 5]
```diff
@@ -16,23 +16,214 @@ export interface UseFmeServiceInputs {
...
+        | "schedules/{category}/{name}"
+        | "schedules/{category}/{name}/metrics"
+        | "schedules/{category}/{name}/trigger"
+        | "security/accounts"
+        | "security/accounts"
```
Comment (kenlyon) [inline src/activities/UseFmeService.ts]: Duplicate entry.

File: src/activities/UseFmeService.ts [hunk 6]
```diff
@@ -16,23 +16,214 @@ export interface UseFmeServiceInputs {
...
+        | "security/accounts/{account}"
+        | "security/accounts/{account}/email"
+        | "security/accounts/{account}/enabled"
+        | "security/accounts/{account}/ldap"
+        | "security/accounts/{account}/password"
+        | "security/accounts/{account}/password"
```
Comment (kenlyon) [inline src/activities/UseFmeService.ts]: Duplicate entry.

Comment (kenlyon) [reply src/activities/UseFmeService.ts]: I assume this one was meant to be `security/accounts/{account}/passwordChangeNeeded`, which is missing.

File: src/activities/UseFmeService.ts [hunk 7]
```diff
@@ -16,23 +16,214 @@ export interface UseFmeServiceInputs {
...
+        | "transformations/jobs/id/{jobid}/request"
+        | "transformations/jobs/id/{jobid}/result"
+        | "transformations/jobs/queued"
+        | "transformations/jobs/queued/{jobid}"
+        | "transformations/jobs/running"
+        | "transformations/jobs/running/{jobid}"
+        | "transformations/submit/{repository}/{workspace}"
+        | "transformations/transact/{repository}/{workspace}"
+        | "transformations/transactdata/{repository}/{workspace}"
+        | "user/favorites/workspace"
```
Comment (kenlyon) [inline src/activities/UseFmeService.ts]: This should be `user/favorites/workspaces`. (plural)

File: src/activities/UseFmeService.ts [hunk 8]
```diff
@@ -16,23 +16,214 @@ export interface UseFmeServiceInputs {
...
+        | "user/favorites/workspace/{repository}/{workspace}"
```
Comment (kenlyon) [inline src/activities/UseFmeService.ts]: This should be `user/favorites/workspaces/{repository}/{workspace}`. (plural)

File: src/activities/UseFmeService.ts [hunk 9]
```diff
@@ -41,9 +232,16 @@ export interface UseFmeServiceInputs {
         | "application/json"
         | "multipart/form-data";
     /**
-     * @description The body parameters to pass to the service operation when the method is POST or PUT.
+     * @description The body parameters or object to pass to the service operation when the method is POST or PUT.
      */
     parameters?: string | Record<string, any>;
+    /**
+     * @description Parameters used to substitute {token} values in the path.
```
Comment (kenlyon) [inline src/activities/UseFmeService.ts]: Maybe this would benefit from an example to help illustrate the syntax.

**Review by kenlyon — CHANGES_REQUESTED** [review]
Looks very cool. I like the idea of auto-fill from a list of known urls and then having the ability to pass in the substitution values separately. I think that'll be really useful.

File: src/activities/UseFmeService.ts (hunk 9 above)
Comment (rcooney) [reply src/activities/UseFmeService.ts]: Done

File: src/activities/UseFmeService.ts (hunk 8 above)
Comment (rcooney) [reply src/activities/UseFmeService.ts]: Fixed

File: src/activities/UseFmeService.ts (hunk 7 above)
Comment (rcooney) [reply src/activities/UseFmeService.ts]: Fixed

File: src/activities/UseFmeService.ts (hunk 6 above)
Comment (rcooney) [reply src/activities/UseFmeService.ts]: Fixed

File: src/activities/UseFmeService.ts (hunk 5 above)
Comment (rcooney) [reply src/activities/UseFmeService.ts]: Fixed

File: src/activities/UseFmeService.ts (hunk 4 above)
Comment (rcooney) [reply src/activities/UseFmeService.ts]: Removed

File: src/activities/UseFmeService.ts (hunk 3 above)
Comment (rcooney) [reply src/activities/UseFmeService.ts]: Fixed

File: src/activities/UseFmeService.ts (hunk 2 above)
Comment (rcooney) [reply src/activities/UseFmeService.ts]: Fixed

File: src/activities/UseFmeService.ts (hunk 1 above)
Comment (rcooney) [reply src/activities/UseFmeService.ts:129]: Added

File: src/activities/UseFmeService.ts (hunk 9 above)
Comment (kenlyon) [reply src/activities/UseFmeService.ts]: Nice, thanks!

**Review by kenlyon — APPROVED** [review]

**Pull request merged**